sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/NoNullForCollectionReturnCheck.java

```diff
@@ -181,10 +181,9 @@ public void visitToken(DetailAST detailAST) {
      * @return true, when method return collection.
      */
     private boolean isReturnCollection(DetailAST methodDef) {
-        DetailAST methodType = methodDef.findFirstToken(TokenTypes.TYPE);
-        methodType = methodType.getFirstChild();
-        return methodType.getType() == TokenTypes.ARRAY_DECLARATOR
-                || collectionList.contains(methodType.getText());
+        final DetailAST methodType = methodDef.findFirstToken(TokenTypes.TYPE);
+        final boolean isArrayType = methodType.findFirstToken(TokenTypes.ARRAY_DECLARATOR) != null;
```

Comment:
Is this safe for current sevntu's version of checkstyle or must this rely solely on newest checkstyle?

See https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/sevntu-checks/pom.xml#L18-L19

Reply:
This relies on 8.43 (or whenever main PR is merged)

Reply:
Actually, we should be good to merge this now, let's see what CI says.

Reply:
I would think regression might show more than CI.

Reply:
https://nmancus1.github.io/sevntu-diff-3145-b/index.html

Reply:
I am not following what the new URL is. Both list the same "master" and "patch" commit.

Reply:
The report is using sevntu checks built from this PR.

Reply:
Then how is it different from the one in the first post?

Reply:
The one in the first post is using sevntu checks from the master branch, ran using report generation in https://github.com/checkstyle/checkstyle/pull/9922.

So, this report is ran using sevntu checks built form master, which shows the regression:  https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/24f673f_2021192452/reports/diff/index.html

And this report was generated with the same Checkstyle codebase, but with sevntu checks built from the code in this PR: https://nmancus1.github.io/sevntu-diff-3145-b/index.html , showing that there is no regression once this PR is merged.